src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/ValuesDocument.java

```diff
@@ -27,7 +23,7 @@ public Document build(Set values) throws Exception {
         Element valuesElement = document.createElement("uniqueValues");
         document.appendChild(valuesElement);
 
-        for (String value : encodeValues(values)) {
+        for (String value : new ValueEncoder().encodeValues(values)) {
```

Comment:
Here it might be worth renaming 'value' to be 'encodedValue' (or if you take on my other suggestion, 'serialisedValue').